pkg/sqlxtransaction/sqlxtransaction.go

```diff
@@ -0,0 +1,40 @@
+// Package sqlxtransaction helps executing a function inside an SQLX transaction.
+package sqlxtransaction
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/nexmoinc/gosrvlib/pkg/logging"
+	"go.uber.org/zap"
+)
+
+// ExecFunc is the type of the function to be executed inside a SQL Transaction.
+type ExecFunc func(ctx context.Context, tx *sqlx.Tx) error
+
+// Exec executes the specified function inside a SQL transaction.
+func Exec(ctx context.Context, db *sqlx.DB, run ExecFunc) error {
+	tx, err := db.BeginTxx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("unable to start an SQLX transaction: %w", err)
+	}
+
+	defer func() {
+		if err := tx.Rollback(); err != nil && !errors.Is(err, sql.ErrTxDone) {
```

Comment:
Is it safe to call `tx.Rollback()` after `tx.Commit()` finished successfully? It seems logically incorrect to do so.

Reply:
I think that is safe to do

Reply:
Yes as we check for the sql.ErrTxDone.